src/OpenLoco/Paint/Paint.cpp

```diff
@@ -41,10 +42,128 @@ namespace OpenLoco::Paint
         return &_session;
     }
 
+    void registerHooks()
+    {
+        registerHook(
+            0x004622A2,
+            [](registers& regs) -> uint8_t {
+                registers backup = regs;
+
+                PaintSession session;
+                session.generate();
+
+                regs = backup;
+                return 0;
+            });
+    }
+
+    // 0x00461CF8
+    static void tilePaintSetup(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x00461CF8, regs);
+    }
+
+    // 0x004617C6
+    static void tilePaintSetup2(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x004617C6, regs);
+    }
+
+    // 0x0046FA88
+    static void entityPaintSetup(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x0046FA88, regs);
+    }
+
+    // 0x0046FB67
+    static void entityPaintSetup2(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x0046FB67, regs);
+    }
+
+    template<uint8_t rotation>
+    void generateRotated(PaintSession& session)
+    {
+        auto* dpi = session.dpi();
+        uint16_t numVerticalQuadrants = (dpi->height + (rotation == 0 ? 1040 : 1056)) >> 5;
```

Comment:
Any idea where 1040 and 1056 come from?

Reply:
No idea, i might check rct2 and see if they exist

Reply:
rct2 had similar (although not in OpenRCT2). I've no idea as to why though

Reply:
hmm.... Feel like it might have to do with the max tile height. Can you add a // TODO for finding the meaning of these constants?